commit a32c7075cf634aee1f4f3deecd27f194097ec0c2
Author: Jacob Walls <jacobtylerwalls@gmail.com>
Date:   Wed Apr 1 09:36:16 2026 -0400

    Fixed #37017 -- Fixed setting or clearing of request.user after alogin/alogout().
    
    Regression in 31a43c571f4d036827d4fd7a5f615591637dc1be.

diff --git a/django/contrib/auth/__init__.py b/django/contrib/auth/__init__.py
index 2702c38aa4..21e6dc43d6 100644
--- a/django/contrib/auth/__init__.py
+++ b/django/contrib/auth/__init__.py
@@ -205,6 +205,8 @@ async def alogin(request, user, backend=None):
     await request.session.aset(SESSION_KEY, user._meta.pk.value_to_string(user))
     await request.session.aset(BACKEND_SESSION_KEY, backend)
     await request.session.aset(HASH_SESSION_KEY, session_auth_hash)
+    if hasattr(request, "user"):
+        request.user = user
     if hasattr(request, "auser"):
 
         async def auser():
@@ -244,13 +246,21 @@ async def alogout(request):
             user = None
     await user_logged_out.asend(sender=user.__class__, request=request, user=user)
     await request.session.aflush()
-    if hasattr(request, "auser"):
+
+    has_user = hasattr(request, "user")
+    has_auser = hasattr(request, "auser")
+    if has_user or has_auser:
         from django.contrib.auth.models import AnonymousUser
 
-        async def auser():
-            return AnonymousUser()
+        anon = AnonymousUser()
+        if has_user:
+            request.user = anon
+        if has_auser:
 
-        request.auser = auser
+            async def auser():
+                return anon
+
+            request.auser = auser
 
 
 def get_user_model():
